src/rgw/rgw_file.cc

```diff
@@ -1159,6 +1159,15 @@ namespace rgw {
       }
     }
 
+    int overlap = 0;
+    if ((static_cast<off_t>(off) < f->write_req->real_ofs) &&
+        (static_cast<long>(f->write_req->real_ofs - off) < sysconf(_SC_PAGESIZE))) {
```

Comment:
of course, the client's pagesize (like its mount options) is not knowable, right?  if we unconditionally attempt this correction, do we do just as well?